src/_utils.js

```diff
@@ -28,12 +28,30 @@ export const addClassName = (path, jsxId) => {
   // Find className and collect spreads
   for (let i = attributes.length - 1, attr; (attr = attributes[i]); i--) {
     const node = attr.node
+
     if (t.isJSXSpreadAttribute(attr)) {
+      if (t.isObjectExpression(node.argument)) {
+        const properties = node.argument.properties
+
+        const index = properties.findIndex(
+          property => property.key.name === 'className'
+        )
+
+        if (~index) {
+          className = attr.get('argument').get(`properties.${index}`)
+
+          // Remove jsx spread attribute if there is only className property
+          if (node.argument.properties.length === 1) {
+            attr.remove()
+          }
+          break
+        }
+        continue
+      }
+
       const name = node.argument.name
       const attrNameDotClassName = t.memberExpression(
-        t.isMemberExpression(node.argument)
-          ? node.argument
-          : t.identifier(name),
+        t.isIdentifier(node.argument) ? t.identifier(name) : node.argument,
```

Comment:
Sorry I had a second thought, we should probably not handle/support the function call and just let this fail. The reason being that we cannot make assumptions about the function and be sure that it is idempotent i.e. not sure if calling it multiple times  returns always the same thing. I would discourage inline calls like `{...foo()}`